src/couch_event/src/couch_event_server.erl

```diff
@@ -74,68 +66,218 @@ handle_cast(Msg, St) ->
     couch_log:notice("~s ignoring cast ~w", [?MODULE, Msg]),
     {noreply, St}.
 
-handle_info({'DOWN', Ref, process, Pid, _Reason}, St) ->
-    case khash:get(St#st.by_pid, Pid) of
-        {Ref, OldDbNames} ->
-            unregister(St, Pid, OldDbNames);
-        undefined ->
-            ok
-    end,
-    {noreply, St};
+handle_info({'DOWN', Ref, process, Pid, _Reason}, #st{by_pid = ByPid} = St) ->
+    #{Pid := {Ref, OldDbNames}} = ByPid,
```

Comment:
Note: here we strengthen the state logic, asserting that we never expect to get a 'DOWN' from an unknown process. Previously, we were a bit too defensive expecting when we handled `undefined` in the `khash:get(St#st.by_pid, Pid)` result.

When a new process is registered, we always monitor it. When an existing process is found, we don't re-monitor but re-use the previous monitor.